src/global_scheduler/global_scheduler_algorithm.cc

```diff
@@ -24,9 +24,10 @@ void GlobalSchedulerPolicyState_free(GlobalSchedulerPolicyState *policy_state) {
  */
 bool constraints_satisfied_hard(const LocalScheduler *scheduler,
                                 const TaskSpec *spec) {
-  if (scheduler->info.static_resources.at("CPU") == 0) {
+  if (scheduler->info.static_resources.count("CPU") == 1 &&
+      scheduler->info.static_resources.at("CPU") == 0) {
```

Comment:
Should we add an extra check here that it's only for actor creation tasks?

Reply:
what would be the reason for a special check? I encourage to not special case anything. We need to have a common abstraction that works for both tasks and actors. Both can be thought of as resource consumers, and the scheduler's job is to meet resource demand with resource supply. Once the correct common abstraction is identified, there should be no need to special case.

Reply:
Probably no need for a check here. However, the reason would be that currently actor creation tasks are the only tasks where the task resource requirements at run time differ from the task resource requirements during scheduling (because during scheduling we need to be aware of subsequent method resource requirements).